source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,184 @@
+=======
+Indexes
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol   
+
+Overview
+--------
+
+Indexes support the efficient execution of queries in MongoDB. Without
+indexes, MongoDB must scan *every* document in a collection to find the
+documents that match each query. Collection scans are slow and can
+negatively affect the performance of your application. When an
+appropriate index exists for a query, MongoDB uses that index to limit
+the number of documents scanned, thus improving query performance. 
+
+The :manual:`indexes section </indexes>` of the MongoDB manual provides a
+detailed overview of indexes in MongoDB.
+
+Query Coverage and Performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you execute a query against MongoDB, your query can include three
+parts:
+
+- query criteria that specify field(s) and value(s) you are looking for,
+- options that affect the query's execution (e.g. read concern),
+
+- (optional) projection criteria to specify the fields MongoDB should
+  return
+
+When all the fields specified in the query criteria and projection of a
+query are indexed, MongoDB returns results directly from the index
+without scanning any documents or bringing documents into memory. Queries
+that use indexes are more performant than queries that rely on
+collection scans.
+
+The MongoDB manual addresses both :manual:`query coverage
+</core/query-optimization/#read-operations-covered-query>` and
+:manual:`index intersection </core/index-intersection>` (using multiple
+indexes to fulfil a query) in more detail.
+
+Operational Considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To improve query performance, you should build indexes on fields that
+appear often in queries and for all operations that involve sorted
+results. Each index that you add consumes disk space and memory
+when active so you should track index
+memory and disk usage for capacity planning. Adding indexes also has a
+negative performance impact for write operations. When designing your data model and choosing the indexes your application requires, you must weigh the benefits and impacts of each index that you add.
+
+:manual:`Indexing Strategies </applications/indexes>` and :manual:`Data Modeling and Indexes </core/data-model-operations/#data-model-indexes>` provide additional insight into index behaviors and considerations.
+
+Index Types
+-----------
+
+MongoDB supports a number of different index types to support querying
+your data. The following sections describe the most common index types
+and provide sample code that creates each index type.
+
+Single Field and Compound Indexes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Single field indexes are user-defined ascending or descending indexes on
```

Comment:
linking to a definition of ascending or descending might help here -- otherwise I think we might be better off without it, since the direction doesn't matter in 99% of cases.

Reply:
Even though you mention it below, it's still kind of confusing in my opinion because *why* an index needs a direction is unclear.